frontend/src/hooks/usePersistentState.ts

```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect, SetStateAction } from 'react';
+
+export const usePersistentState = <T extends any = any>(
+  key: string,
+  defaultValue?: any
+): [T, React.Dispatch<SetStateAction<T>>] => {
+  const [state, setState] = useState<T>(() => {
+    const existing = localStorage.getItem(key);
+    return existing ? JSON.parse(existing) : defaultValue;
```

Comment:
I run into this error when switching from an existing crossfeed installation to this branch:

![image](https://user-images.githubusercontent.com/1689183/92978965-4bdbfc00-f45f-11ea-9ef1-0a1b04120d0e.png)

![image](https://user-images.githubusercontent.com/1689183/92978972-51d1dd00-f45f-11ea-8e45-09e4293b1af9.png)


I assume this is because `localStorage.getItem("token")` gives me a non-JSON-stringified version, because it was set using an older version of the code, so running `JSON.parse` on it gives an error.

Could we instead fail silently and return an empty value?